std/assembly/string.ts

```diff
@@ -127,10 +210,10 @@ export class String {
     var leftLength = left.length;
     if (leftLength != right.length) return false;
 
-    return !memory.compare(
-      changetype<usize>(left) + HEADER_SIZE,
-      changetype<usize>(right) + HEADER_SIZE,
-      (<usize>leftLength << 1)
+    return !compareUTF16(
```

Comment:
This makes me think: The only thing in question here is equality, so an optimized general comparison should be faster than explicitly comparing UTF16 I assume. However, the `memcpy` we currently have does just a byte-per-byte comparison, so it isn't. So I wonder whether attempting to implement an optimized memcmp (i.e. 32-bit aligned chunks where possible, similar to the optimizations in memcpy) for just equality might make sense?

Reply:
Yah this makes sense for non-shinkage opt level for maximum performance. And I also thought about it. But for a start point it would be nice to have at least this=)

Reply:
Yeah, it's already an improvement as you implemented it. I guess we can think about the other things later :)